src/main/groovy/com/jetbrains/python/envs/PythonEnvsPlugin.groovy

```diff
@@ -83,51 +83,118 @@ class PythonEnvsPlugin implements Plugin<Project> {
+
+                File temporaryUnzipFolder = new File(project.buildDir, "python-build-tmp")
+                project.ant.unzip(src: pythonBuildZip, dest: temporaryUnzipFolder)
+                project.exec {
+                    commandLine "bash", new File(temporaryUnzipFolder, "install.sh")
+                    environment PREFIX: installDir
+                }
+
+                temporaryUnzipFolder.deleteDir()
+                pythonBuildZip.delete()
+            }
+        }
+    }
 
-            outputs.dir(jythonBootstrapDir)
+    private static File findCondaExecutable(File condaDir) {
+        return new File(condaDir, Os.isFamily(Os.FAMILY_WINDOWS) ? "Scripts/conda.exe" : "bin/conda")
+    }
 
+    private Task createPythonEnvUnix(Project project, PythonEnv env, File envDir) {
+        return project.tasks.create(name: "Create $env.type interpreter '$env.name'") {
             onlyIf {
-                !jythonBootstrapDir.exists()
+                !envDir.exists()
             }
 
+            dependsOn "install_python_build"
+
             doLast {
-                project.javaexec {
-                    main = '-jar'
-                    args conf.singleFile, '-s', '-d', jythonBootstrapDir, '-t', 'standard'
+                try {
+                    project.exec {
+                        executable new File(project.buildDir, "python-build/bin/python-build")
+                        args env.version, envDir
+                    }
+                }
+                catch (Exception e) {
+                    printRedText(e.message)
+                    throw new StopExecutionException()
                 }
 
-                project.exec {
-                    executable new File(jythonBootstrapDir, "bin/pip")
-                    args "install", "virtualenv"
+                pipInstall(project, envDir, env.packages)
+            }
+        }
+    }
+
+    private Task createPythonEnvWindows(Project project, PythonEnv env, File envDir){
+        return project.tasks.create(name: "Create $env.type interpreter '$env.name'") {
+            onlyIf {
+                !envDir.exists()
+            }
+
+            doLast {
+                try {
+                    String urlToMsi = "https://www.python.org/ftp/python/$version/python-$version.msi"
+                    project.ant.get(dest: new File(project.buildDir, "python-$version.msi")) {
+                        url(url: urlToMsi)
+                    }
+                }
+                catch (Exception e) {
+                    printRedText(e.message)
+                    throw new StopExecutionException()
                 }
+
+                // TODO msi install script
+
+                pipInstall(project, envDir, env.packages)
             }
         }
     }
 
-    def findCondaExecutable(root) {
-        if (root.endsWith("/")) {
-            root = root.substring(0, root.length() - 1)
+    private Task createJythonEnv(Project project, PythonEnv env, File envDir) {
```

Comment:
Is it Jython virtualenv creation or just bootstraping main Jython interpreter?  Before those were 2 separate steps. Why is it changed?